### ISSUE TITLE
problem with db="-"

### ISSUE THREAD

**Issue by scafoglieri** [jmoiron/sqlx#388]
Hi,
I don't understand what's happens with this use case. I've a struct like this:

```
type Person struct {
	ID             uint64    `json:"id" db:"id"`
	Createdatetime time.Time `json:"-" db:"-"`
	Name           string    `json:"name" db:"name"`
	Surname        string    `json:"surname" db:"surname"`
	Age            int       `json:"age" db:"age"`
	Old            bool      `json:"old" db:"old"`
}
```

which is filled with data taken from a sqlite3 database:

```
db, err_db := sqlx.Connect("sqlite3", DB_NAME)
defer db.Close()
...
p := Person{}
query := "SELECT * FROM person LIMIT 1"
err_i := db.Get(&p, query)
...
```

Nothing special here, but I always get this error:

"missing destination name createdatetime in *mypkg.Person"

Ok, I'm not using Unsafe() method, but I know that db="-" ignores the related attribute in the Person struct.
In fact, if I comment lines 769-771 from sqlx.go I get the 'correct Person' without 'Createdatetime'.

But if I set

`db = db.Unsafe()`

than whatever I put in db inline tag (e.g. `json:"-" db:"mmmkdkd"`), that field is always ignored. This is, in fact, "unsafe".
So now, what's the correct way to skip attributes with db="-", but without using Unsafe()?

Thanks!

**Comment by jmoiron** [comment]
`db.Unsafe()` allows you to partially scan results;  it's there to allow you to provide a "partial destination" for query results, and to also mirror the behaviours of some of the standard library unmarshallers.

It's not the default because this can lead to very subtle runtime errors when you misspell `db` struct tags.  FWIW, by default, you shouldn't need any of your `db` struct tags to support Person.  Conversely, if you do not want to set `createdatetime`, you're expected to not be `SELECT`-ing it from the database;  this is somewhat natural from Go's `x var declared but not used` rules.

tldr;

* unsafe allows you to ignore the fact that you are not scanning all data coming from the db into your destination type, to prevent typos from sneaking into production
* the correct way to skip attributes without using `Unsafe` is to not request them from the database.

**Comment by ptman** [comment]
I have a problem that I believe db:"-" struct tag would solve

For a struct like

```
type Foo struct {
    MetadataJSON types.NullJSONText `db:"metadata"`
    Metadata map[string]interface{} // can't use db:"-" to omit
}
f := &Foo{}
tx.Get(f, `SELECT metadata FROM foo LIMIT 1`)
```

gives

```
sql: Scan error on column index 6, name "metadata": unsupported Scan, storing driver.Value type <nil> into type *map[string]interface {}
```

@jmoiron would you kindly advice?

**Comment by ptman** [comment]
Well, I solved it with `db:"ignore"`, but I think supporting db:"-" would be cleaner
